src/clout/core.cljc

```diff
@@ -2,24 +2,36 @@
   "A small language for routing."
   (:require [clojure.string :as string]
             [clojure.set :as set]
-            [instaparse.core :as insta]))
+            [instaparse.core :as insta]
+            #?(:cljs [clout.internal.regex :refer [re-unicode-letter]])))
 
 (def ^:private re-chars (set "\\.*+|?()[]{}$^"))
+#?(:cljs (def ^:private re-char-escapes
+           (into {} (for [c re-chars] [c (str "\\" c)]))))
 
 (defn- re-escape [s]
-  (string/escape s #(if (re-chars %) (str \\ %))))
-
-(defn- re-groups* [^java.util.regex.Matcher matcher]
-  (for [i (range (.groupCount matcher))]
-    (.group matcher (int (inc i)))))
+  (string/escape s #?(:clj #(if (re-chars %) (str \\ %)) :cljs re-char-escapes)))
+
+(defn- re-match-groups [re s]
+  {:pre [(string? s)]}
```

Comment:
I think I'd prefer that we mimic `re-matches` for both clj and cljs in this case, so rather than add a precondition, can you instead keep the cljs error message?